File: dotnet/src/Buble.Sdk/BubleClientOptions.cs
namespace Buble.Sdk;

/// <summary>
/// Configuration used to create a <see cref="BubleClient"/>.
/// </summary>
public sealed class BubleClientOptions
{
    /// <summary>
    /// Buble API key. When omitted, the client reads BUBLE_API_KEY.
    /// </summary>
    public string? ApiKey { get; set; }

    /// <summary>
    /// API base URL. Defaults to https://buble.ai, or BUBLE_BASE_URL when set.
    /// </summary>
    public string? BaseUrl { get; set; }

    /// <summary>
    /// Request timeout. Defaults to 60 seconds.
    /// </summary>
    public TimeSpan? Timeout { get; set; }

    /// <summary>
    /// Optional externally managed HTTP client.
    /// </summary>
    public HttpClient? HttpClient { get; set; }

    /// <summary>
    /// Extra headers applied to every request. Per-request headers override these.
    /// </summary>
    public IDictionary<string, string> Headers { get; } = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
}

File: swift/Sources/Buble/BubleClientOptions.swift
/// Configuration used to create a `BubleClient`.
public struct BubleClientOptions: Sendable {
    public static let defaultBaseURL = URL(string: "https://buble.ai")!

    public var apiKey: String?
    public var baseURL: URL?
    public var timeout: TimeInterval
    public var headers: [String: String]
    public var transport: HTTPTransport?

    public init(
        apiKey: String? = nil,
        baseURL: URL? = nil,
        timeout: TimeInterval = 60,
        headers: [String: String] = [:],
        transport: HTTPTransport? = nil
    ) {
        self.apiKey = apiKey
        self.baseURL = baseURL
        self.timeout = timeout
        self.headers = headers
        self.transport = transport
    }
}
